lib/taiko.js

```diff
@@ -1945,7 +1948,7 @@ function dropDown(attrValuePairs, ...args) {
             function selectBox(value) {
                 let found_value = false;
                 for (var i = 0; i < this.options.length; i++) {
-                    if (this.options[i].text === value || this.options[i].value === value) {
+                    if (this.options[i].text === value || this.options[i].value === value || this.options[i].index === value) {
```

Comment:
when an index is given we dont have to iterate that many times, index can be set directly